core/src/engine/Hamiltonian_Heisenberg.cu

```diff
@@ -25,50 +25,74 @@ using Engine::Vectormath::cu_tupel_from_idx;
 namespace Engine
 {
     // Construct a Heisenberg Hamiltonian with pairs
-    Hamiltonian_Heisenberg::Hamiltonian_Heisenberg(
-        scalar external_field_magnitude, Vector3 external_field_normal,
-        intfield anisotropy_indices, scalarfield anisotropy_magnitudes, vectorfield anisotropy_normals,
-        pairfield exchange_pairs, scalarfield exchange_magnitudes,
-        pairfield dmi_pairs, scalarfield dmi_magnitudes, vectorfield dmi_normals,
-        DDI_Method ddi_method, intfield ddi_n_periodic_images, bool ddi_pb_zero_padding, scalar ddi_radius,
-        quadrupletfield quadruplets, scalarfield quadruplet_magnitudes,
-        std::shared_ptr<Data::Geometry> geometry,
-        intfield boundary_conditions
-    ) :
-        Hamiltonian(boundary_conditions),
-        geometry(geometry),
-        external_field_magnitude(external_field_magnitude * C::mu_B), external_field_normal(external_field_normal),
-        anisotropy_indices(anisotropy_indices), anisotropy_magnitudes(anisotropy_magnitudes), anisotropy_normals(anisotropy_normals),
-        exchange_pairs_in(exchange_pairs), exchange_magnitudes_in(exchange_magnitudes), exchange_shell_magnitudes(0),
-        dmi_pairs_in(dmi_pairs), dmi_magnitudes_in(dmi_magnitudes), dmi_normals_in(dmi_normals), dmi_shell_magnitudes(0), dmi_shell_chirality(0),
-        quadruplets(quadruplets), quadruplet_magnitudes(quadruplet_magnitudes),
-        ddi_method(ddi_method), ddi_n_periodic_images(ddi_n_periodic_images), ddi_pb_zero_padding(ddi_pb_zero_padding), ddi_cutoff_radius(ddi_radius),
-        fft_plan_reverse(FFT::FFT_Plan()), fft_plan_spins(FFT::FFT_Plan())
-    {
-        // Generate interaction pairs, constants etc.
-        this->Update_Interactions();
+Hamiltonian_Heisenberg::Hamiltonian_Heisenberg(
+    scalar external_field_magnitude, Vector3 external_field_normal, intfield anisotropy_indices,
+    scalarfield anisotropy_magnitudes, vectorfield anisotropy_normals, scalarfield cubic_anisotropy_magnitudes,
```

Comment:
Looks like you're missing the cubic_anisotropy_indices here

Reply:
Thanks, I've managed to compile with CUDA. I hopefully fixed everything so that the cubic anisotropy should now work with CUDA. Note that I had to also include the fix proposed in https://github.com/spirit-code/spirit/issues/602 since otherwise I couldn't compile with CUDA 11.